src/blockchain/chain.rs

```diff
@@ -69,11 +106,31 @@ impl State<Mockchain> {
 }
 
 impl Blockchain<Mockchain> {
-    pub fn new(genesis_data: GenesisData) -> Self {
-        let state = State::new(&genesis_data);
+    pub fn new(genesis_data: GenesisData, storage: &std::path::PathBuf) -> Self {
+        let mut state = State::new(&genesis_data);
+
+        std::fs::create_dir_all(storage).unwrap();
+        let mut sqlite = storage.clone();
+        sqlite.push("blocks.sqlite");
+
+        let storage = SQLiteBlockStore::new(sqlite.to_str().unwrap());
+
+        if let Some(tip_hash) = storage.get_tag(LOCAL_BLOCKCHAIN_TIP_TAG).unwrap() {
+            debug!("restoring state at tip {}", tip_hash);
+
+            // FIXME: should restore from serialized chain state once we have it.
+            for info in storage
+                .iterate_range(&<Mockchain as BlockConfig>::BlockHash::zero(), &tip_hash)
+                .unwrap()
```

Comment:
We might want to start introducing proper error handling instead of an `unwrap` bonanza everywhere. Will this go away as the FIXME comment above suggests, or should the initialization code change to returning `Result`?

Reply:
Yeah, we should introduce proper error handling, but that probably should be a separate PR. Also as you say this code will probably end up getting rewritten/dropped anyway when we have proper chain state restoration.